pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -239,6 +243,13 @@ func getKubeVPAClient(timeout time.Duration) (vpa.Interface, error) {
 	return vpa.NewForConfig(clientConfig)
 }
 
+// VPAInformerFactory vpai.SharedInformerFactory
+func getVPAInformerFactory(client vpa.Interface) (vpai.SharedInformerFactory, error) {
+	// default to 300s
+	resyncPeriodSeconds := time.Duration(config.Datadog.GetInt64("kubernetes_informers_resync_period"))
+	return vpai.NewSharedInformerFactory(client, resyncPeriodSeconds*time.Second), nil
+}
+
```

Comment:
I definitely want some 👀 on this. Since VPAs are not built in, we couldn't follow the same pattern. I believe I needed to add a VPA factory outside of the collector code.

Reply:
Adding VPA factory makes sense to me but I think we should move this part to  https://github.com/DataDog/datadog-agent/blob/fb92594cf8692f46cdbe3cac4b08fd2d8a3a0da6/pkg/util/kubernetes/apiserver/apiserver.go#L357-L375
and perhaps keep the same structure as other factory
```suggestion
// VPAInformerFactory vpai.SharedInformerFactory
func getVPAInformerFactory() (vpai.SharedInformerFactory, error) {
	resyncPeriodSeconds := time.Duration(config.Datadog.GetInt64("kubernetes_informers_resync_period"))
        client, err := getKubeVPAClient(0) // No timeout for the Informers, to allow long watch.
        if err != nil {
		log.Errorf("Could not get apiserver client: %v", err)
		return nil, err
	}
	return vpai.NewSharedInformerFactory(client, resyncPeriodSeconds*time.Second), nil
}

```